### PR TITLE
Project cards on dashboard powered by (mock) HTTP 

### PR DESCRIPTION
**Pull request by seanknox**
Foundational work to support fetching project card data from Brigade. The service and components make real requests over HTTP. At the moment I'm intercepting HTTP request with mock data using Angular's `in-memory-web-api` which is useful for dev only. Subsequent work will allow overriding with a real external Brigade API.

- Introduces `ProjectService` to fetch project data from the Brigade API and provide to the dashboard component. 
- Adds momentjs support

Related to #129

### CODE REVIEW

**Comment by flynnduism** [comment]
Holler at me if rebasing `dashboard.component.html` is weird as a result of the icon edits!

**Comment by seanknox** [comment]
@flynnduism implemented setting the project status classes dynamically and to keep it simple, I opted to use an icon for an active build instead of the `rolling.png`, at least for now. We can revisit of course. Also added an icon for if a build is in an "unknown" state.

kashti_and_1__seanknox_naga____go_src_github_com_azure_kashti__zsh__and__users_seanknox_go_src_github_com_azure_kashti__git_

Going to merge this in to call dashboard layout "done", with hooking up to real data coming next.

File: src/app/dashboard/dashboard.component.html [hunk 1]
```diff
@@ -1,28 +1,20 @@
...
-      'project-active': projectsbuild.lastBuild.worker.status == 'Active'
-    }">
-      <a href="">
-
+  <section class="column" *ngFor="let project of projects">
+    <div class="project">
+      <a routerLink="/project/{{project.id}}">
         <div class="status">
-          <div class="success"><i class="icon ion-md-checkmark-circle status-success"></i></div>
+          <i [ngClass]="calculateStatusClasses(project.lastBuild)"></i>
```
Comment (seanknox) [inline src/app/dashboard/dashboard.component.html]: @flynnduism this calls the `calculateStatusClasses` function in the dashboard component to figure out which classes to apply to the icon element.

File: src/app/dashboard/dashboard.component.ts [hunk 2]
```diff
@@ -7,9 +11,61 @@ import { Component, OnInit } from '@angular/core';
...
+
+    switch (status) {
+      case 'Success':
+        return this.successStateClasses();
+      case 'Failed':
+        return this.failureStateClasses();
+      case 'Active':
+        return this.activeStateClasses();
+      default:
+        return this.unknownStateClasses();
```
Comment (seanknox) [inline src/app/dashboard/dashboard.component.ts]: The class calc function itself. It determines what CSS classes to apply to each project status icon, e.g. "passed", "failed", "active", "unknown. Not very elegant, as requires passing in a lastBuild object, fetching the worker status, and comparing the status strings. I wonder if we could simplify the data model on the Brigade API. It'd be nice to replace this switch with object literals, using the status value to construct the appropriate css class name, e.g. `project-failed`. Still need to apply a corresponding icon, however.

cc @technosophos

File: package.json [hunk 3]
```diff
@@ -22,6 +22,8 @@
     "@angular/platform-browser": "^5.0.0",
     "@angular/platform-browser-dynamic": "^5.0.0",
     "@angular/router": "^5.0.0",
+    "angular-in-memory-web-api": "^0.5.2",
```
Comment (johnpapa) [inline package.json]: This should likely be a dev dependency. right?

File: src/app/app.module.ts [hunk 4]
```diff
@@ -18,10 +24,19 @@ import { FooterComponent } from './footer/footer.component';
...
   imports: [
+    MomentModule,
     BrowserModule,
+    HttpClientModule,
     AppRoutingModule,
+
+    // The HttpClientInMemoryWebApiModule module intercepts HTTP requests
+    // and returns simulated server responses.
+    // Remove it when a real server is ready to receive requests.
+    HttpClientInMemoryWebApiModule.forRoot(InMemoryDataService, {
```
Comment (johnpapa) [inline src/app/app.module.ts]: could be a dev time line of code

File: src/app/project.service.ts [hunk 5]
```diff
@@ -0,0 +1,45 @@
+import { Injectable } from '@angular/core';
+import { Observable } from 'rxjs/Observable';
+import { of } from 'rxjs/observable/of';
+import { catchError, map, tap } from 'rxjs/operators';
+
+import { HttpClient, HttpHeaders } from '@angular/common/http';
+
+import { Project } from './models/project';
+
+const httpOptions = {
```
Comment (johnpapa) [inline src/app/project.service.ts]: would be cool to encaspulate this outside, or do it in  service configuration code. a class or interceptor

File: src/app/project.service.ts [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+      );
+  }
+
+  /**
+   * Handle Http operation that failed.
+   * Let the app continue.
+   * @param operation - name of the operation that failed
+   * @param result - optional value to return as the observable result
+   */
+  private handleError<T>(operation = 'operation', result?: T) {
```
Comment (johnpapa) [inline src/app/project.service.ts]: considering abstracting this for all http services.

File: package.json (hunk 3 above)
Comment (seanknox) [reply package.json]: Yeah, that's probably the right call. I was using it as a stand-in for backend API endpoint, but having it available in the dev context only seems to be a good thing to be explicit with.

File: src/app/app.module.ts (hunk 4 above)
Comment (seanknox) [reply src/app/app.module.ts]: What did you mean by this?

File: src/app/project.service.ts (hunk 5 above)
Comment (seanknox) [reply src/app/project.service.ts]: Agreed! Will tackle that in future refactoring.

File: src/app/project.service.ts (hunk 6 above)
Comment (seanknox) [reply src/app/project.service.ts]: Good call.

**Pull request merged**